fpga.chart.py

```diff
@@ -181,26 +195,24 @@ def _get_xilinx_fpga_power(self, idx):
 
     def _get_fpga_temp(self):
         while True:
-            time.sleep(self.temp_power_update_interval - 2)
+            before = time.time()
             for idx in range(self.fpga_count):
-                if os.path.exists(self.intel_cmd) and os.access(self.intel_cmd, os.X_OK):
-                    self.temp[idx] = self._get_intel_fpga_temp(idx)
-                    continue
-                if os.path.exists(self.xilinx_cmd) and os.access(self.xilinx_cmd, os.X_OK):
-                    self.temp[idx] = self._get_xilinx_fpga_temp(idx)
-                    continue
+                self.temp[idx] = self.get_fpga_temp[self.manufacturer](idx)
+            after = time.time()
+            time_taken = after - before
+
+            time.sleep(self.temp_power_update_interval - time_taken)
```

Comment:
Looks good, only caveat is what happens if this becomes negative?

Reply:
It would fail